File: src/main/java/iteration2/Instructor.java
package iteration2;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import iteration2.Course;
import iteration2.FacultyMember;
import iteration2.GivenCourse;
import iteration2.Student;
import org.apache.log4j.LogManager;
import org.apache.log4j.Logger;

import java.util.Random;
import java.util.ArrayList;


public class Instructor extends FacultyMember {
	final Logger logger = LogManager.getLogger(Instructor.class);

	private ArrayList<iteration2.Course> coursesOfferedList;

	public Instructor(String name, String id) {
		super(name, id);
		coursesOfferedList = new ArrayList<>();
	}



	public void gradeStudents(Student student, Course course) {
		Random rand = new Random();
		float int_random = Math.max( rand.nextInt(9)/2.0f,Math.min(rand.nextInt(9)/2.0f,rand.nextInt(9)/2.0f));
		logger.info("Graded student "+ student.getName() + " for course : " + course.getName() + " grade is : " + int_random);
		student.getStudentSemester().getGivenCourses().add(new GivenCourse(course.getCode(),int_random,course.getCredit()));
	}

	public ArrayList<iteration2.Course> getCoursesOfferedList() {
		return coursesOfferedList;
	}

	public void setCoursesOfferedList(ArrayList<Course> coursesOfferedList) {
		this.coursesOfferedList = coursesOfferedList;
	}



}

File: Instructor.py
import random
from typing import List
import logging
from Course import Course
from GivenCourse import GivenCourse
from Person import Person


class Instructor(Person):
    def __init__(self, name: str, id: str):
        super().__init__(name, id)
        self.courses_offered_list = []

    def grade_students(self, student, course):
        try:
            int_random = max(random.randint(0, 8), min(random.randint(0, 8), random.randint(0, 8))) / 2
            # int_random = max(random.uniform(0, 4.5), min(random.uniform(0, 4.5), random.uniform(0, 4.5)))
            logging.info(f"Graded student {student.name} for course {course.name} grade is: {int_random}")
            student.student_semester.given_courses.append(GivenCourse(course.code, int_random, course.credit))
        except AttributeError:
            print("Error: student or course not found")